support-workers/src/main/scala/com/gu/support/workers/lambdas/Handler.scala

```diff
@@ -10,43 +10,42 @@ import io.circe.{Decoder, Encoder}
 import scala.concurrent.duration._
 import scala.concurrent.{Await, ExecutionContext, Future}
 
-abstract class Handler[T, R](implicit decoder: Decoder[T], encoder: Encoder[R]) extends RequestStreamHandler {
+abstract class Handler[T, R](
+  implicit
+  decoder: Decoder[T],
+  encoder: Encoder[R],
+  ec: ExecutionContext
+) extends RequestStreamHandler {
 
   import com.gu.support.workers.encoding.Encoding._
 
   type HandlerResult = (R, RequestInfo)
 
-  def HandlerResult(r: R, ri: RequestInfo): HandlerResult = (r, ri) // scalastyle:ignore method.name
-
-  protected def handler(input: T, error: Option[ExecutionError], requestInfo: RequestInfo, context: Context): HandlerResult
+  type FutureHandlerResult = Future[HandlerResult]
 
-  def handleRequest(is: InputStream, os: OutputStream, context: Context): Unit =
-    try {
-      in(is).flatMap {
-        case (i, error, requestInfo) =>
-          val result = handler(i, error, requestInfo, context)
-          out(result._1, result._2, os)
-      }.get
-    } catch ErrorHandler.handleException
-}
+  def HandlerResult(r: R, ri: RequestInfo): HandlerResult = (r, ri) // scalastyle:ignore method.name
```

Comment:
Separate from your change, but this should really be a case class

Reply:
yes I almost did that, then decided it might be too cluttered and resolved to do it next time!  Same for the tuple from `in` as well.  I might go ahead anyway now that you've spotted it too 😁